### ISSUE TITLE
Saving a model with a custom primary key

### ISSUE THREAD

**Issue by bbedward** [tortoise/tortoise-orm#242]
Hi, I'm not sure if this is an issue or I'm doing something wrong but I picked tortoise for a new asyncio project (previously have been using peewee) and have encountered 1 issue.

I have a model such as:

```
class User(Model):
    id = fields.BigIntField(pk=True)
    name = fields.CharField(max_length=50)
    created_at = fields.DatetimeField(auto_now_add=True)

    class Meta:
        table = "users"
```

When I create an instance of it, doing something like:

```
user = User(
   id = 34553535,
   name = 'username'
)
await user.save()
```

There's no error/exception/or any feedback, and the record is not inserted into the database. I know with peewee I'd have to do something like `.save(force_insert=True)` - but didn't see anything in the tortoise documentation that would achieve a similar result.

Also, is there a way to tell if the save() is successful? With peewee I'd be able do something like

```
if user.save() > 0:
   print("User saved successfully")
else:
   print("User did not save")
```

Thanks for the work on this plugin, I've been looking for an asyncio ORM and was happy to find one under such active development.

**Comment by bbedward** [comment]
Adding generated=False to the field solved the not inserting problem:

```
class User(Model):
    id = fields.BigIntField(pk=True, generated=False)
    name = fields.CharField(max_length=50)
    created_at = fields.DatetimeField(auto_now_add=True)
    modified_at = fields.DatetimeField(auto_now=True)
```

However, I'm still a little alarmed that the `.save()` call raised no exception, if that's something I'm doing wrong please let me know.

**Comment by AEnterprise** [comment]
it's a bit counter intuitive but if you do not add the ``generated=False`` thing it ignores whatever ID you pass it and gens a new one

probably caused it to think your object already existed in the database and did an update SQL query instead of an insert

**Comment by grigi** [comment]
Ok, I had a look at what was happening.
If you don't set the `generated=False`, since it is a int-PK, it assumes that we are dealing with an autonumber. 
It then always fetches the ID from the database on first save. So you'll find that after saving the PK has been changed to the provided one by the DB.

This behaviour isn't entierly invalid, but not ideal.

Would it be ok to instead generate a `ValueError` if you try to set a generated field as part of the constructor? (At least it won't be confusing, instead of just ignoring it)

The other question is should we support a case where we have an AUTONUMBER pk, and someone wants to overwrite record 102, and does a `Model(id=102, ...).save()` ? I can think of many bad reasons to do this, but not any good ones?
One should rather do a `Model.get(id=102) ...` or a `Model.filter(id=102).update(...)` ?

**Comment by grigi** [comment]
I added a check for the upcoming 0.15.0 that will stop you from passing a value that is going to be ignored. I hope that will help resolve some confusion.

**Comment by bbedward** [comment]
I think raising a `ValueError` when trying to override a generated field is good. I'm ok with the current behavior otherwise.

Thanks again, really enjoying tortoise so far.

**Comment by AEnterprise** [comment]
maybe save could raise an error (or log a warning) if no rows where affected?

personally can't think of any cases where you'd be calling save but it being ok if nothing is actually saved for some reason

**Comment by grigi** [comment]
I just released v0.15.0 Please test to see if it does it right :-)

**Comment by bbedward** [comment]
Just tried 0.15.0 and it raised the exception, going to close the issue.
